common/include/pcl/PCLPointField.h

```diff
@@ -52,4 +50,11 @@ namespace pcl
     s << "  " << v.count << std::endl;
     return (s);
   }
+
+  inline bool operator==(const PCLPointField& f1, const PCLPointField& f2)
```

Comment:
Having a member function makes the overload space smaller. Is there any reason for choosing a free function?

Also, I noticed you were adding noexcept to function definitions. Maybe add a noexcept here

Reply:
No particular reason, just a [good practice](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-symmetric)

Reply:
I don't know what subtle differences would arise when both classes are same, arguments are const (and function is const). Could you please expand on that a bit.

Reply:
There is not much to expand here. It is just a common pattern to implement operators (symmetric or asymmetric) as free function, specially if they don't need to access private members of the class ( ``friend`` is another thing to avoid ). It enforces encapsulation. It does not pollute encapsulation, for sure you do not want to declare another ``operator==`` for the same types. So, as long as this operator applies as a general case and it is not an equality function only for a specific case, this free function operator is just fine.